### ISSUE TITLE
bug in pose/datasets/mpii.py  and  pose/losses/jointsmseloss.py 

### ISSUE THREAD

**Issue by mengyingfei** [bearpaw/pytorch-pose#93]
Q1.in pose/datasets/mpii.py
`        for i in range(nparts):
            # if tpts[i, 2] > 0: # This is evil!!
            if tpts[i, 1] > 0:
                tpts[i, 0:2] = to_torch(transform(tpts[i, 0:2]+1, c, s, [self.out_res, self.out_res], rot=r))
                target[i], vis = draw_labelmap(target[i], tpts[i]-1, self.sigma, type=self.label_type)
                target_weight[i, 0] *= vis`
I think it should be  if tpts[i, 2] > 0 not if tpts[i, 1] >  0,  tpts[i, 2] indicates if the key point is visible. 
The file of coco.py writes like this.

Q2.in pose/losses/jointsmseloss.py 
`        for idx in range(num_joints):
            heatmap_pred = heatmaps_pred[idx].squeeze()
            heatmap_gt = heatmaps_gt[idx].squeeze()
            if self.use_target_weight:
                loss += 0.5 * self.criterion(
                    heatmap_pred.mul(target_weight[:, idx]),
                    heatmap_gt.mul(target_weight[:, idx])
                )
            else:
                loss += 0.5 * self.criterion(heatmap_pred, heatmap_gt)`
target_weight[:, idx] should be target_weight[idx, 0]. Because target_weight[idx, 0] indicates if the key point is visible.
I don't know if I said it right, please take a look, thanks.

**Comment by Malefikus** [comment]
> Q1.in pose/datasets/mpii.py
> 
> ![1212](https://user-images.githubusercontent.com/43380407/55677738-cd9d3100-591f-11e9-869e-8a2c1c161cf4.png)
> 
> I think it should be if tpts[i, 2] > 0 not if tpts[i, 1] > 0, tpts[i, 2] indicates if the key point is visible.
> The file of coco.py writes like this.
> 
> Q2.in pose/losses/jointsmseloss.py
> 
> ![Selection_044](https://user-images.githubusercontent.com/43380407/55677741-f0c7e080-591f-11e9-8c1f-1882d33aa4ee.png)
> 
> target_weight[:, idx] should be target_weight[idx, 0]. Because target_weight[idx, 0] indicates if the key point is visible.
> I don't know if I said it right, please take a look, thanks.

Hi, I happen to browse the same parts of the code and my considerations are that:

For Q1, the "if" statement is no longer used to judge if the point is visible (you can simply delete the line). The original usage of the if statement is to create masked heatmaps for non-visible points, but here the publisher creates unmasked heatmaps for all of the joints by default, and implement training-time masking via the target_weight indicator in jointsmseloss. In test time, we should not mask the target heatmaps because the occluded joints are still required for calculating the accuracy.

For Q2, the target_weight is now a batch_size*njoints*1 tensor, and target_weight[ :, idx ] indicates the visibility of this joint for all of the samples in one batch.

**Comment by jaideep11061982** [comment]
what is njoints here do you have an idea. 
In my model i have to regress for 6 coordinates and 1 confidence mask. 
Is 6 be njoins in this case ?
